pkg/network/common/dns.go

```diff
@@ -139,31 +139,32 @@ func (d *DNS) Update(dns string) (bool, error) {
 	defer d.lock.Unlock()
 
 	if err != nil {
-		klog.Errorf("Unable to resolv %q: %v", dns, err)
-		err = d.updateNextQueryTime(dns)
+		d.updateNextQueryTime(dns)
 		return false, err
 	}
 
-	return d.updateDNSValue(dns, ips, ttl)
+	changed := d.updateDNSValue(dns, ips, ttl)
+	return changed, nil
 }
 
-func (d *DNS) updateNextQueryTime(dns string) error {
+func (d *DNS) updateNextQueryTime(dns string) {
 	res, ok := d.dnsMap[dns]
 	if !ok {
 		// Should not happen, all operations on dnsMap are synchronized by d.lock
-		return fmt.Errorf("DNS value not found in dnsMap for domain: %q", dns)
+		klog.Errorf("DNS value not found in dnsMap for domain: %q", dns)
+	} else {
```

Comment:
would be better to `return` rather than putting the rest of the function in an `else`